app/src/test/java/com/courseproject/tindar/usecases/likelist/LikeListInteractorUnitTest.java

```diff
@@ -0,0 +1,111 @@
+
+        @Override
+        public boolean checkLiked(String MOCK_USER_ID, String MOCK_OTHER_USER_ID) {
+            return isLiked;
+        }
+
+        @Override
+        public void addToMatched(String MOCK_USER_ID, String MOCK_OTHER_USER_ID) {
+            matchList.add(MOCK_USER_ID + " is matched " + MOCK_OTHER_USER_ID);
+        }
+
+        @Override
+        public void addLike(String MOCK_USER_ID, String MOCK_OTHER_USER_ID) {
+            likeList.add(MOCK_USER_ID + " likes " + MOCK_OTHER_USER_ID);
+        }
+
+        @Override
+        public void removeLike(String MOCK_USER_ID, String MOCK_OTHER_USER_ID) {
+            likeList.remove(MOCK_USER_ID + " likes " + MOCK_OTHER_USER_ID);
+        }
+
+        @Override
+        public void removeFromMatched(String MOCK_USER_ID, String MOCK_OTHER_USER_ID) {
+            matchList.remove(MOCK_USER_ID + " unmatched with " + MOCK_OTHER_USER_ID);
+        }
+    }
+    @Test
+    public void addLikeAndMatched() {
+        /** Test that USER_ID_1 and USER_ID_2 are matched when both users "like" each other **/
+        ArrayList<String> matchList = new ArrayList<>();
+        ArrayList<String> likeList = new ArrayList<>();
+        LikeListDsGateway likeListDsGateway = new MockLikeListDsGateway(true, matchList,
+                likeList);
+        LikeListInteractor likeListInteractor = new LikeListInteractor(likeListDsGateway);
+        likeListInteractor.addLike(USER_ID_1, USER_ID_2);
+        assertEquals(USER_ID_1 + " likes " + USER_ID_2, likeList.get(0));
+        assertEquals(USER_ID_1 + " is matched " + USER_ID_2, matchList.get(0));
+    }
+
+    @Test
+    public void addLikeAndNotMatched() {
+        /** Test that USER_ID_1 is added to USER_ID_2 likeList, but not match the users. **/
+
+        ArrayList<String> matchList = new ArrayList<>();
+        ArrayList<String> likeList = new ArrayList<>();
+        LikeListDsGateway likeListDsGateway = new MockLikeListDsGateway(false, matchList,
+                likeList);
+        LikeListInteractor likeListInteractor = new LikeListInteractor(likeListDsGateway);
+        likeListInteractor.addLike(USER_ID_1, USER_ID_2);
+        assertEquals(USER_ID_1 + " likes " + USER_ID_2, likeList.get(0));
+        assertTrue(matchList.isEmpty());
+    }
+
+    @Test
+    public void removeLikeAndMatched() {
+        /** Test that MOCK_USER_ID unlikes MOCK_OTHER_USER_ID, and that the users are also
+         * unmatched. **/
+        ArrayList<String> matchList = new ArrayList<>();
+        ArrayList<String> likeList = new ArrayList<>();
+        LikeListDsGateway likeListDsGateway = new MockLikeListDsGateway(true, matchList,
+                likeList);
+        LikeListInteractor likeListInteractor = new LikeListInteractor(likeListDsGateway);
+        likeListInteractor.addLike(USER_ID_1, USER_ID_2);
+        assertEquals(USER_ID_1 + " likes " + USER_ID_2, likeList.get(0));
+        assertEquals(USER_ID_1 + " is matched " + USER_ID_2, matchList.get(0));
```

Comment:
wait shouldn't this only happen if likeList already has USER_ID_2 likes USER_ID_1? in that case shouldn't the line above be at index 1?